colabs/preprocessing.ipynb

```diff
@@ -0,0 +1,284 @@
+        "# unzip it\n",
+        "!tar xf spark-3.2.1-bin-hadoop3.2.tgz\n",
+        "# install findspark\n",
+        "!pip install -q findspark\n",
+        "# clone github repo\n",
+        "!git clone https://github.com/PiotrMaciejKowalski/BigData2022-films\n",
+        "# Przeniesienie plików z BigData2022-films do katalogu nadrzędnego\n",
+        "!mv BigData2022-films/* .\n",
+        "!mv BigData2022-films/.* .\n",
+        "!rmdir BigData2022-films"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 3,
+      "metadata": {
+        "id": "fG3Xpjc4MvYY"
+      },
+      "outputs": [],
+      "source": [
+        "import os\n",
+        "\n",
+        "# setup environment variables for our Spark Session to work\n",
+        "os.environ[\"JAVA_HOME\"] = \"/usr/lib/jvm/java-8-openjdk-amd64\"\n",
+        "os.environ[\"SPARK_HOME\"] = '/content/spark-3.2.1-bin-hadoop3.2'\n",
+        "\n",
+        "from lib.pyspark_startup import init, load"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 4,
+      "metadata": {
+        "id": "lIJChpFeBN-U"
+      },
+      "outputs": [],
+      "source": [
+        "spark = init()"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 31,
+      "metadata": {
+        "id": "fOLm43gvBWNM"
+      },
+      "outputs": [],
+      "source": [
+        "# Ładowanie danych z dysku google\n",
+        "path = \"/content/drive/.shortcut-targets-by-id/1VcOir9FMG8LzEsUE-Q8YA79c_sV0tJwp/bigdata2022/\"\n",
+        "\n",
+        "df = spark.read.parquet(path + \"clean_df.parquet\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Preprocessing - przyłożenie stworzonych funkcji"
+      ],
+      "metadata": {
+        "id": "yXU7xq4qUiNk"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from lib.pyspark_preprocesing import one_hot_encoding\n",
+        "\n",
+        "df = one_hot_encoding(df, [\"rodzaj_produkcji\",\"gatunek\"])"
+      ],
+      "metadata": {
+        "id": "3I-HEm45a0wo"
+      },
+      "execution_count": 32,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from lib.feature_creators import add_epoch_column\n",
+        "\n",
+        "df = add_epoch_column(df)"
+      ],
+      "metadata": {
+        "id": "Fd-TvW73a_9U"
+      },
+      "execution_count": 33,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from lib.film_people_list import people_film_merge_columns\n",
+        "\n",
+        "dataCollect = df.collect()\n",
+        "df = people_film_merge_columns(df,df['id'], add_column = True)"
+      ],
+      "metadata": {
+        "id": "U_ebpfN6W4RN"
+      },
+      "execution_count": 34,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "df.show()"
```

Comment:
Show() do usunięcia, jako że to jest skrypt przetwarzający, to podgląd jest zbędny.

Reply:
usunąć

Reply:
Usunięte